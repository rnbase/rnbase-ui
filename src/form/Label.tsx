import React from 'react';
import { StyleSheet, Text, TextProps } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface LabelProps extends TextProps {
  children: React.ReactNode;
}

const Label: React.FC<Themed<typeof createStyleSheet, LabelProps>> = ({
  theme: { styles },
  children,
  style,
  ...props
}) => (
  <Text {...props} style={[styles.root, style]} numberOfLines={1}>
    {children}
  </Text>
);

const createStyleSheet = ({ colors, fonts }: Theme) =>
  StyleSheet.create({
    root: {
      ...fonts.normal,
      height: 16,
      fontSize: 16,
      lineHeight: 16,
      flexShrink: 1,
      marginRight: 20,
      color: colors.gray,
    },
  });

export default withTheme(Label, createStyleSheet, 'Form.Label');
