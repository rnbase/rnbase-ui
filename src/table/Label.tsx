import React from 'react';
import { StyleSheet, Text, TextProps } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface LabelProps extends TextProps {
  children: React.ReactNode;
}

const Label: React.FC<Themed<typeof createStyleSheet, LabelProps>> = ({
  theme: { styles },
  children,
  style,
  ...props
}) => (
  <Text {...props} style={[styles.root, style]} numberOfLines={1}>
    {children}
  </Text>
);

const createStyleSheet = ({ colors, fonts }: Theme) =>
  StyleSheet.create({
    root: {
      ...fonts.normal,
      fontSize: 17,
      flexShrink: 1,
      marginRight: 20,
      color: colors.black,
    },
  });

export default withTheme(Label, createStyleSheet, 'Form.Label');
