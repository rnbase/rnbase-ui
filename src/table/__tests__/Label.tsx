import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

import Label from '../Label';

type Props = React.ComponentProps<typeof Label>;

/**
 * Under test
 */
const createElement = (props: Partial<Props>) => <Label {...props}>label</Label>;

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally', () => {
  expect(createRenderer({})).toMatchSnapshot();
});

it('should render with custom styles', () => {
  expect(createRenderer({ style: { color: 'red' } })).toMatchSnapshot();
});
