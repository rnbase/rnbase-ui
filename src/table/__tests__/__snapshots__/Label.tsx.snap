// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render normally 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "flexShrink": 1,
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "400",
        "includeFontPadding": false,
        "marginRight": 20,
        "textAlignVertical": "center",
      },
      undefined,
    ]
  }
>
  label
</Text>
`;

exports[`should render with custom styles 1`] = `
<Text
  numberOfLines={1}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "flexShrink": 1,
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "400",
        "includeFontPadding": false,
        "marginRight": 20,
        "textAlignVertical": "center",
      },
      Object {
        "color": "red",
      },
    ]
  }
>
  label
</Text>
`;
